src/Microsoft.AspNetCore.SignalR/HubEndPoint.cs

```diff
@@ -59,24 +59,54 @@ public class HubEndPoint<THub, TClient> : IInvocationBinder where THub : Hub<TCl
 
         public async Task OnConnectedAsync(ConnectionContext connection)
         {
-            await ProcessNegotiate(connection);
+            var output = Channel.CreateUnbounded<byte[]>();
+            var connectionContext = new HubConnectionContext(output, connection);
+
+            await ProcessNegotiate(connectionContext);
```

Comment:
Our client could (3rd party clients might decide not to) but it only feels like a convenience - IMHO the server should always reject invalid requests and not rely on the client.